lib/r10k/action/deploy/environment.rb

```diff
@@ -43,6 +44,34 @@ def call
 
         private
 
+        def extract_credentials
+          if @sshkey_path && @token_path
+            raise R10K::Error, "Cannot specify both an SSH key and a token to use with this deploy."
+          end
+
+          if @sshkey_path
+            path = convert_dash_to_stdin(@sshkey_path)
+            if File.exists?(path)
+              # TODO if this is reading from stdin, write the key to a (password
+              # protected?) tempfile instead and save that path for passing down
+              @settings[:sshkey_file] = path
+            else
+              raise R10K::Error, _("{%path} does not exist, cannot load SSH key") % { path: path }
+            end
+          elsif @token_path
+            path = convert_dash_to_stdin(@sshkey_path)
+            if File.exists?(path)
+              @settings[:token] = File.read(path)
+            else
+              raise R10K::Error, _("{%path} does not exist, cannot load OAuth token") % { path: path }
+            end
+          end
+        end
+
+        def convert_dash_to_stdin(path)
+          path == '-' ? '/dev/stdin' : path
```

Comment:
Do we need to support Windows?? I don't think that `/dev/stdin` works on Windows, however I do think that reading the IO object `STDIN`/`$stdin` would be a portable way to handle that.

Reply:
I think we probably do.